In [3]:
# excercise 3: Write a program that will give you the sum of 3 digits
def sum_digit(n):
    m = str(n)
    tong = 0
    for i in m:
        tong = tong + int(i)
    print(tong)

In [5]:
sum_digit(567)

18


In [6]:
# excercise 4: Write a program that will reverse a 4 DIGIT number-Also it checks whether the reverse is true.
def reverse_num(n):
    m = str(n)[::-1]   # [start:stop:step]
    print(m) 


In [7]:
reverse_num(1231)

1321


In [ ]:
# An Armstrong number is one whose sum of digits raised to the power three equals the
# number itself. 371, for example, is an Armstrong number because + + I* *3
# 371.

In [35]:
def Armstrong_num(n):
    lst = []
    for i in range(0, n+1):
        m = str(i)
        tong = 0
        for j in m:
            tong = int(j)**3 + tong
        if tong == i:
            lst.append(i)
    return(lst)
            

In [36]:
print("then amstrong number are: ", Armstrong_num(100000))

then amstrong number are:  [0, 1, 153, 370, 371, 407]


In [46]:
# split the name
ten = input('nhap vao cac ten ngan cach boi space')
ten_lit = ten.split(" ")
order = 0
for i in ten_lit:
    order = order + 1
    print('ten nguoi so ', order, " la", str(i))

ten nguoi so  1  la dsf
ten nguoi so  2  la sdf
ten nguoi so  3  la sdfsd
ten nguoi so  4  la e
ten nguoi so  5  la tewr
ten nguoi so  6  la fsd
ten nguoi so  7  la fdv
ten nguoi so  8  la cxvx
ten nguoi so  9  la af
ten nguoi so  10  la asf


In [47]:
# given 2 list, select odd num from first list and even in second list
def fiter_lst(n,m):
    final_lst = []
    for i in n:
        if i%2 != 0:
            final_lst.append(i)
    for j in m:
        if j%2 == 0:
            final_lst.append(j)
    return final_lst


In [48]:
print(fiter_lst([1,2,3,4,5,6], [1,2,3,4,5,6]))

[1, 3, 5, 2, 4, 6]


In [53]:
def past_pop(n):
    order = 11
    m = n*1.1
    for i in range(10, 0, -1):
        order = order - 1
        m = m/1.1
        print('year', order, 'has population of', m)

In [54]:
past_pop(10000)

year 10 has population of 10000.0
year 9 has population of 9090.90909090909
year 8 has population of 8264.462809917353
year 7 has population of 7513.148009015775
year 6 has population of 6830.134553650703
year 5 has population of 6209.213230591548
year 4 has population of 5644.739300537771
year 3 has population of 5131.5811823070635
year 2 has population of 4665.07380209733
year 1 has population of 4240.976183724845


**Exercise 24: Check Palindrome Number Write a program to check if the given number is a palindrome number.**

*A palindrome number is a number that is same after reverse. For example 545, is the palindrome numbers*



In [6]:
def check_pali_num(n):
    m = str(n)[::-1]
    print(f"the reverse num is {m}")
    if int(m) == int(n):
        print(f'{n} is a palindrome numbers')
    else:
        print(f'{n} is not a palindrome numbers')

In [8]:
check_pali_num(234)

the reverse num is 432
234 is not a palindrome numbers


In [19]:
# Exercise 27: Print multiplication table form 1 to 10
for i in range(1,11):
    for j in range(1,11):
        print(i*j, end = " ")
    print("\n")

1 2 3 4 5 6 7 8 9 10 

2 4 6 8 10 12 14 16 18 20 

3 6 9 12 15 18 21 24 27 30 

4 8 12 16 20 24 28 32 36 40 

5 10 15 20 25 30 35 40 45 50 

6 12 18 24 30 36 42 48 54 60 

7 14 21 28 35 42 49 56 63 70 

8 16 24 32 40 48 56 64 72 80 

9 18 27 36 45 54 63 72 81 90 

10 20 30 40 50 60 70 80 90 100 



**print a triagle**

In [31]:
n = int(input('nhập vào độ lớn tam giác: '))
for i in range (1,n+1):
    for j in range(i, i+1):
        print("* "*j)

* 
* * 
* * * 
* * * * 
* * * * * 


**excercise 32: User will provide 2 numbers you have to find the HCF of those 2 numbers**

In [37]:
x = int(input('nhập vào số đầu tiên: '))
y = int(input('nhập vào số thứ hai: '))
lst_x = []
lst_y = []
lst_c = []
inex = -1
for i in range(1,x+1):
    if x%i == 0:
        lst_x.append(i)
for j in range(1,y+1):
    if y%j == 0:
        lst_y.append(j)
for l in lst_x:
    for m in lst_y:
        if l == m:
            lst_c.append(l)
for k in lst_c:
    inex += 1
    if lst_c[inex] > lst_c[0]:
        lst_c[0] = lst_c[inex]
print(f"các ước số thứ nhất là {lst_x}")
print(f"các ước số thứ hai là {lst_y}")
print(f"ước số chung lớn nhất là {lst_c[0]}")

các ước số thứ nhất là [1, 5]
các ước số thứ hai là [1, 2, 4, 5, 10, 20]
ước số chung lớn nhất là 5


**Exercise 62: Find the sum of the series upto n terms Write a program to calculate the sum of series up to n term. For example, if n =5 the series will become 2 + 22 + 222 + 2222 + 22222 = 24690**

In [7]:
def sum_n_term(n,x):
    tong = 0
    for i in range(1,n+1):
        tong = tong + int(i*str(x))
    print(tong)


In [8]:
sum_n_term(5,2)

24690


**Exercise 66: Print the following pattern Write a program to print the following start pattern using the for loop**

In [17]:
n = int(input('nhap vao do lon tam giac: '))
for i in range(1, n+1):
    print(i*"* ")
for j in range(n-1,0,-1):
    print(j*"* ")

* 
* * 
* * * 
* * * * 
* * * * * 
* * * * 
* * * 
* * 
* 


**Excercise 67: Write a program that keeps on accepting a number from the user until the user enters Zero. Display the sum and average of all the numbers.**

In [22]:
tong = 0
count = 0
m = int(input('nhap vao so m: '))
while m != 0:
    count = count + 1
    tong = tong + m
    m = int(input('nhap vao so m: '))
print(f"tong cac so la {tong}")
print(f"trung binh cac so la {tong/count}")



tong cac so la 56
trung binh cac so la 56.0


**excercise 68: Write a program to print the first 25 odd numbers**

In [26]:
n = int(input('nhap vao so luong so le ban muon: '))
lst = []
tong = 0
while len(lst) < n:
    tong = tong + 1
    if tong % 2 !=0:
        lst.append(tong)
print(lst)
    

[1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49, 51, 53, 55, 57, 59, 61, 63, 65, 67, 69, 71, 73, 75, 77, 79, 81, 83, 85, 87, 89, 91, 93, 95, 97, 99, 101, 103, 105, 107, 109, 111, 113, 115, 117, 119, 121, 123, 125, 127, 129, 131, 133, 135, 137, 139, 141, 143, 145, 147, 149, 151, 153, 155, 157, 159, 161, 163, 165, 167, 169, 171, 173, 175, 177, 179, 181, 183, 185, 187, 189, 191, 193, 195, 197, 199, 201, 203, 205, 207, 209, 211, 213, 215, 217, 219, 221, 223, 225, 227, 229, 231, 233, 235, 237, 239, 241, 243, 245, 247, 249, 251, 253, 255, 257, 259, 261, 263, 265, 267, 269, 271, 273, 275, 277, 279, 281, 283, 285, 287, 289, 291, 293, 295, 297, 299, 301, 303, 305, 307, 309, 311, 313, 315, 317, 319, 321, 323, 325, 327, 329, 331, 333, 335, 337, 339, 341, 343, 345, 347, 349, 351, 353, 355, 357, 359, 361, 363, 365, 367, 369, 371, 373, 375, 377, 379, 381, 383, 385, 387, 389, 391, 393, 395, 397, 399, 401, 403, 405, 407, 409, 411, 413, 415, 417, 419, 421,

**Exercise 69: Write a function called exponent(base, exp) that returns an int value of base raises to the power of exp. Note here exp is a non-negative integer, and the base is an integer.**

In [1]:
def exponent(base, exp):
    tong = 0
    result = 1
    while tong < (exp):
        tong = tong + 1
        result = result*base
    print(result)

In [4]:
exponent(2, 4)

16


**Excercise 82: Print the first 20 numbers of a Fibonacci series**

In [10]:
def finbonaci(n):
    lst = [0,1]
    tong = 0
    while len(lst) < n:
        tong +=1
        if tong == lst[-1] + lst[-2]:
            lst.append(tong)
    print(lst)


In [11]:
finbonaci(12)

[0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89]


**Exercise 84: Print characters from a string that are present at an even index number Write a program to accept a string from the user and display characters that are present at an even index number.**

For example, str = "pynative" so you should display ‘p’, ‘n’, ‘t’, ‘v’.


In [15]:
m = str(input('nhap vao cac ky tu: '))
lst = []
for i in range(0, int(len(m)),2):
    lst.append(m[i])
print(m)
print(lst)

pynative
['p', 'n', 't', 'v']


**Exercise 1B: Create a string made of the middle three characters**

In [17]:
m = str(input('nhap vao chu: '))
middle = len(m)//2
result = m[middle - 1] + m[middle] +m[middle + 1]
print(result)

con


**Exercise 1B: Create a string made of the middle n characters**

In [38]:
def laysogiua(m,n):
    middle = len(m)//2
    result = ""
    for i in range(middle - n//2, middle):
        result = result + m[i]
    for j in range(middle, middle + n//2):
        result = result + m[j]
    print(result)

In [46]:
laysogiua('123123huytrangds123123',10)

huytrangds


**Excercise 88:Find the index position of a particular character in another string.**

In [47]:
m = str(input('input string: '))
n = str(input('string to find: '))
result = m.index(n)
print(result)

2


**Exercise 92: Arrange string characters such that lowercase letters should come first**

Given string contains a combination of the lower and upper case letters. Write a program to arrange the
characters of a string so that all lowercase letters should come first.


In [70]:
def arrange_str(n):
    lst_lower = []
    lst_upper = []
    for i in n:
        if i.islower():
            lst_lower.append(i)
        else:
            lst_upper.append(i)
    final = lst_lower + lst_upper
    print(''.join(final))

In [71]:
arrange_str('HUYjfahaUJHNj')


jfahajHUYUJHN


**Exercise 96: Find all occurrences of a substring in a given string by ignoring the case Write a program to find
all occurrences of “USA” in a given string ignoring the case.**

str1 = "Welcome to USA. usa awesome, isn't it?"
expected ans --> USA:-->2**

In [78]:
def countstr(m,n):
    m_lower = m.lower()
    n_lower = n.lower()
    print(m_lower.count(n_lower))

In [80]:
countstr("Welcome to USA. usa awesome, isn't it?", "USA")

2


**Excercise 97 Write a python program to convert a string to title case without using the title()**

In [97]:
m = str(input('nhap chu: '))
result = ""
if m[0] == " ":
    result+=m[0]
elif m[0] != " ":
    result+=m[0].upper()
for i in range(1, len(m)):
    if m[i-1] == " " and m[i] != " ":
        result+=m[i].upper()
    else:
        result += m[i]
print(result)


Hom Nay La Mot Ngay Dep Troi


In [98]:
a=input("enter the title: ")
b=a.split()
r=''
print(b)
for i in b:
 r=r+i.capitalize()+" "
 
print(r)

['hom', 'nay', 'la', 'mot', 'ngay', 'dep', 'troi']
Hom Nay La Mot Ngay Dep Troi 


**Exercise 76: Create an inner function to calculate the addition in the following way**

Create an outer function that will accept two parameters, a and b

Create an inner function inside an outer function that will calculate the addition of a and b

At last, an outer function will add 5 into addition and return it

In [2]:
def outerfunction(m,n):
    def innerfunction(m,n):
        return m+n
    add = innerfunction(m,n) + 5
    return add

In [3]:
print(outerfunction(6,5))

16


**Exercise 77: Create a recursive function**

A Write a program to create a recursive function to calculate the sum of numbers from 0 to 10.

A recursive function is a function that calls itself again and again.

Expected Output:
55

In [4]:
def recursive_sum(n):
    if n == 0:
        return 0
    else:
        return n + recursive_sum(n-1)

result = recursive_sum(10)
print(result)


55


In [5]:
recursive_sum(10)

55

**B) Write a program to create a recursive function to calculate the factorial of numbers from 0 to 10.**

A recursive function is a function that calls itself again and again.

Expected Output:
120

In [11]:
def fac_num(n):
    if n == 1:
        return 1
    else:
        return n*fac_num(n-1)

In [13]:
print(fac_num(5))

120


**Exercise 101: Split a string on hyphens**

Write a program to split a given string on hyphens and display each substring.

str1 = Emma-is-a-data-scientist

Expected-->Displaying each substring

Emma

is

a

data

scientist


In [17]:
m = str(input('nhap vao chuoi:'))
n = m.split("-")
for i in range(0, len(n)):
    print(n[i])

Emma
is
a
data
scientist


**Exercise 102: Remove empty strings from a list of strings**

In [23]:
m = ["Emma", "Jon", "", "Kelly", None, "Eric", ""]
result = []
for i in m:
    if i != "" and isinstance(i, str) :
        result.append(i)
print(result)

['Emma', 'Jon', 'Kelly', 'Eric']


**Exercise 105: Find words with both alphabets and numbers**

Write a program to find words with both alphabets and numbers from an input string

In [12]:
def extract_word(m):
    n = m.split(" ")
    result = []
    for k in n:
        def extract_word(k):
            if not any(i.isalpha() for i in k):
                return False
            if not any(i.isdigit() for i in k):
                return False
            return True
        if extract_word(k):
            result.append(k)
    return result

In [13]:
print(extract_word("Emma253 is Data scientist50000 and AI Expert a4234"))

['Emma253', 'scientist50000', 'a4234']


**Exercise 106: Replace each special symbol with # in the following string**

str1 = '/*Jon is @developer & musician!!'

Expected-->##Jon is #developer # musician##

In [79]:
def hide_special(m):
    n = m.split(" ")
    result = []
    for i in n:
        k = []
        for j in i:
            if (j.isalpha() or j.isdigit()):
                k.append(j)
            else:
                k.append('#')
        result_2 = "".join(k)
        result.append(result_2)
    return result
input_pdf_path = "D:/Hutson/Learning/Python/HOMEWORK BUỔI 9.pdf"
output_pdf_path = "D:/Hutson/Learning/Python/HOMEWORK BUỔI 9 test.pdf"


In [45]:
" ".join(hide_special('/*Jon is @developer & musician!!'))

'##Jon is #developer # musician##'

In [48]:
sentence = "/*Jon is @developer & musician!!"
import re
clean_sentence=re.sub('[^A-Za-z0-9\s]+',"#",sentence)
print(clean_sentence)

#Jon is #developer # musician#


In [66]:
sample_list = [34, 54, 67, 89, 11, 43, 94]
index_4 = sample_list[4]
sample_list.remove(index_4)  # Remove the element at index 4
sample_list.insert(2,index_4)
print(sample_list)

[34, 54, 11, 67, 89, 43, 94]


**Exercise 112: Count the occurrence of each element from a list**

Write a program to iterate a given list and count the occurrence of each element and create a dictionary to
show the count of each element.

sample_list = [11, 45, 8, 11, 23, 45, 23, 45, 89]

Expected Output:--> Printing count of each item {11: 2, 45: 3, 8: 1, 23: 2, 89: 1}


In [75]:
list1 = [11, 45, 8, 11, 23, 45, 23, 45, 89]
dict_lst = {}
for i in list1:
    if i in dict_lst:
        dict_lst[i] +=1
    else:
        dict_lst[i] = 1
print(dict_lst)

{11: 2, 45: 3, 8: 1, 23: 2, 89: 1}


**Exercise 116: Iterate a given list and check if a given element exists as a key’s value in a dictionary. If not,
delete it from the list**

roll_number = [47, 64, 69, 37, 76, 83, 95, 97]

sample_dict = {'Jhon':47, 'Emma':69, 'Kelly':76, 'Jason':97}

Expected result:--> After removing unwanted elements from list [47, 69, 76, 97]


In [106]:
roll_number = [47, 64, 69, 37, 76, 83, 95, 97]

sample_dict = {'Emma':69, 'Kelly':76, 'Jason':97, 'mic':6}

result = []
for i in roll_number:
    if i in sample_dict.values():
        result.append(i)
print(result)

[69, 76, 97]


**Exercise 117: Get all values from the dictionary and add them to a list but don’t add duplicates**

speed = {'jan': 47, 'feb': 52, 'march': 47, 'April': 44, 'May': 52, 'June': 53, 'july': 54, 'Aug': 44, 'Sept': 54}

expected_output: [47, 52, 44, 53, 54]


In [110]:
speed = {'jan': 47, 'feb': 52, 'march': 47, 'April': 44, 'May': 52, 'June': 53, 'july': 54, 'Aug': 44, 'Sept': 54}
result = []
for i in speed.values():
    if i not in result:
        result.append(i)
print(result)

[47, 52, 44, 53, 54]


**Exercise 118: Remove duplicates from a list and create a tuple and find the minimum and maximum number**

sample_list = [87, 45, 41, 65, 94, 41, 99, 94]

Expected-->

unique items [87, 45, 41, 65, 99]

tuple (87, 45, 41, 65, 99)

min: 41

max: 99

In [114]:
sample_list = [87, 45, 41, 65, 94, 41, 99, 94]
sample_tuple = tuple(sample_list)
print(sample_tuple)
print(max(sample_tuple))
print(min(sample_tuple))

(87, 45, 41, 65, 94, 41, 99, 94)
99
41


**Exercise 135: Convert two lists into a dictionary**

Below are the two lists. Write a Python program to convert them into a dictionary in a way that item from list1
is the key and item from list2 is the value

keys = ['Ten', 'Twenty', 'Thirty']

values = [10, 20, 30]

Expected output:
{'Ten': 10, 'Twenty': 20, 'Thirty': 30}

In [124]:
keys = ['Ten', 'Twenty', 'Thirty']
values = [10, 20, 30]
my_dict=zip(values,keys)
result=dict(my_dict)
print(result)

{10: 'Ten', 20: 'Twenty', 30: 'Thirty'}


In [136]:
set1 = {10, 20, 30, 40, 50}
set2 = [30, 40, 50, 60, 70]
set3=set1.update(set2)
print(set3)

None
